src/app/health/health-event-dialog.component.ts

```diff
@@ -46,4 +49,18 @@ export class HealthEventDialogComponent implements OnInit {
     this.router.navigate([ 'event', { id: this.data.user, eventId: event._id } ], { relativeTo: this.data.route });
   }
 
+  timer() {
+    const intervalId = setInterval(() => {
+      const time = (300000 - (new Date(Date.now()).getTime() - new Date(this.event.date).getTime())) / 1000;
```

Comment:
Since `setInterval` doesn't run immediately, you could define this function, run it first, and then run it in the `setInterval`.